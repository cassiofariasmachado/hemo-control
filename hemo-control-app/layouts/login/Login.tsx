import React, { useState, FC } from 'react';
import styled from 'styled-components/native';

type LoginProps = {
    onLogin: (username: string, password: string) => void
}

const Container = styled.View`
    flex: 1;
    alignItems: center;
    justifyContent: center;
    backgroundColor: #fff;
`

const Input = styled.TextInput`
    paddingHorizontal: 10px;
    paddingVertical: 10px;
    backgroundColor: #fff;
    alignSelf: stretch;
    marginBottom: 20px;
    marginHorizontal: 30px;
    fontSize: 16px;
    borderBottomWidth: 1px;
    borderColor: black;
`

const LoginButton = styled.TouchableOpacity`
    paddingHorizontal: 20px;
    paddingVertical: 15px;
    borderRadius: 5px;
    backgroundColor: black;
    alignSelf: stretch;
    alignItems: center;
    justifyContent: center;
    marginBottom: 15px;
    marginHorizontal: 30px;
`

const LoginText = styled.Text`
    color: #fff;
    fontWeight: bold;
    fontSize: 16px;
`

export const Login: FC<LoginProps> = ({ onLogin }) => {
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');

    return <>
        <Container>
            <Input placeholder="UsuÃ¡rio" autoCompleteType="username" onChangeText={setUsername} />

            <Input placeholder="Senha" autoCompleteType="password" secureTextEntry={true} onChangeText={setPassword} />

            <LoginButton onPress={() => onLogin(username, password)}>
                <LoginText>Login</LoginText>
            </LoginButton>
        </Container>
    </>;
};
