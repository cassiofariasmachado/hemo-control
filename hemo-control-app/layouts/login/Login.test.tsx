import React from 'react';
import { create } from 'react-test-renderer';
import { Login } from './Login';

describe(Login.name, () => {
    test('renders correctly', () => {
        const onLogin = jest.fn();
        const tree = create(<Login onLogin={onLogin} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
