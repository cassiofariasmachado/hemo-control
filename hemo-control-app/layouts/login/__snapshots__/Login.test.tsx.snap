// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="username"
    onChangeText={[Function]}
    placeholder="UsuÃ¡rio"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "black",
          "fontSize": 16,
          "marginBottom": 20,
          "marginHorizontal": 30,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCompleteType="password"
    onChangeText={[Function]}
    placeholder="Senha"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "black",
          "fontSize": 16,
          "marginBottom": 20,
          "marginHorizontal": 30,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "borderRadius": 5,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 30,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;
